androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/MapsFragment.kt

```diff
@@ -36,8 +45,67 @@ class MapsFragment : Fragment() {
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
+        setupUI()
+        viewModel.state.observe(viewLifecycleOwner, ::onStateChanged)
+    }
+
+    @SuppressLint("MissingPermission")
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<out String>,
+        grantResults: IntArray
+    ) {
+        val permissionResultParser = PermissionResultParser()
+        if (permissionResultParser.isFineLocationPermissionsGranted(permissions, grantResults)) {
+            viewModel.startUpdatingLocation()
+            map.isMyLocationEnabled = true
+        } else {
+            showToast("Location permissions are required to find meals")
+        }
+    }
+
+    private fun setupUI() {
         val mapFragment = childFragmentManager.findFragmentById(R.id.map) as SupportMapFragment?
         mapFragment?.getMapAsync(callback)
     }
+
+    private fun onStateChanged(state: MapsViewModel.State) {
+        when (state) {
+            MapsViewModel.State.LoadingMeals ->
+                Log.i("Map view", "Loading meals")
+            is MapsViewModel.State.LoadedMeals ->
+                onLoadedMeals(state)
+            is MapsViewModel.State.LocationUpdate ->
+                onLocationUpdate(state)
+            is MapsViewModel.State.Failed ->
+                onFailure(state.failure)
+        }
+    }
+
+    private fun onLocationUpdate(state: MapsViewModel.State.LocationUpdate) {
+        map.moveCamera(CameraUpdateFactory.newLatLngZoom(state.userLatLng, 12.0f))
```

Comment:
For now have just left the zoom as a reasonable neighbourhood.  We could consider limiting meals to a certain distance or zooming to bounds in the future.